highcharts-angular/src/lib/highcharts-chart.directive.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Directive,
+  DestroyRef,
+  afterRenderEffect,
+  ElementRef,
+  inject,
+  input,
+  linkedSignal,
+  model,
+  output,
+  OutputEmitterRef,
+  untracked,
+  computed,
+  PLATFORM_ID,
+} from '@angular/core';
+import { isPlatformServer } from '@angular/common';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { HighchartsChartService } from './highcharts-chart.service';
+import { HIGHCHARTS_CONFIG } from './highcharts-chart.token';
+import type { Chart, ChartConstructorType } from './types';
+
+
+@Directive({
+  selector: '[highcharts-chart]',
+})
+export class HighchartsChartDirective {
+  /**
+   * Highcharts library or Highcharts ESM module.
+   * @deprecated
+   */
+  Highcharts = input<Chart['highcharts']>();
+
+  /**
+   * Type of the chart constructor.
+   */
+  constructorType = input<ChartConstructorType>('chart');
+
+  /**
+   * Callback function for the chart.
+   */
+  callbackFunction = input<Highcharts.ChartCallbackFunction>(null);
+
+  /**
+   * When enabled, Updates `series`, `xAxis`, `yAxis`, and `annotations` to match new options.
+   * Items are added/removed as needed. Series with `id`s are matched by `id`;
+   * unmatched items are removed. Omitted `series` leaves existing ones unchanged.
+   */
+  oneToOne = input<boolean>();
+
+  /**
+   * Whether to run the chart outside Angular.
+   * @deprecated
+   */
+  runOutsideAngular = input<boolean>();
+
+  /**
+   * Options for the Highcharts chart.
+   */
+  options = input<Chart['options']>();
+
+  /**
+   * Whether to redraw the chart.
+   * Check how update works in Highcharts
+   * API doc here: https://api.highcharts.com/class-reference/Highcharts.Chart#update
+   */
+  update = model<boolean>();
+
+  chartInstance: OutputEmitterRef<Highcharts.Chart | null> = output<Highcharts.Chart | null>();  // #26
+
+  private destroyRef = inject(DestroyRef);
+
+  private el = inject(ElementRef);
+
+  private platformId = inject(PLATFORM_ID);
+
+  private relativeConfig = inject(HIGHCHARTS_CONFIG, { optional: true });
+
+  private highchartsChartService = inject(HighchartsChartService);
+
+  private highCharts = toSignal(this.highchartsChartService.loaderChanges$);
+
+  private constructorChart = computed<Function>(() => {
+    const constructorType = untracked(this.constructorType);
```

Comment:
If I understand this correctly, it means that changing `[constructorType]` after the very first initialization will have no effect at all. Is this intentional ? should we be able to change the constructor type while during the component life cycle ? :thinking: 

If it can indeed never be changed later on, then I think it should be documented as such. WDYT ?

Reply:
@PowerKiKi , I kept the same behavior as before. If you check the code in [version 4](https://github.com/highcharts/highcharts-angular/blob/ab095ccaa07e5014a1e667506203bcb0c09f4d6a/highcharts-angular/src/lib/highcharts-chart.component.ts#L29), the ngOnChanges method is just tracking the options and update inputs. That’s why I maintained the same behavior. However, I am open to modifying this to track changes whenever this input changes.